cdci_data_analysis/analysis/instrument.py

```diff
@@ -539,48 +553,67 @@ def set_input_products(self, par_dic, input_file_path,input_prod_list_name):
-        if user_catalog_file is not None:
-            catalog_object = load_user_catalog(user_catalog_file)
+        if 'user_catalog_file' in par_dic.keys() and par_dic['user_catalog_file'] is not None:
+            user_catalog_file = par_dic['user_catalog_file']
+            try:
+                catalog_object = load_user_catalog(user_catalog_file)
+            except RuntimeError:
+                raise RequestNotUnderstood('format not valid, a catalog should be provided as a FITS (typical standard OSA catalog) or '
+                                           '<a href=https://docs.astropy.org/en/stable/api/astropy.io.ascii.Ecsv.html>ECSV</a> table.')
             self.set_par('user_catalog', catalog_object)
-            # TODO set selected_catalog ? in this way it will show up in the email url link
-            # and used consistently for the job_id generation
             self.set_par('selected_catalog', json.dumps(catalog_object.get_dictionary()))
-            # TODO verify that the file path is not needed in the frontend (should not)
+            # TODO not needed in the frontend
             par_dic.pop('user_catalog_file', None)
         else:
+            # TODO is this case still in use?
             if 'catalog_selected_objects' in par_dic.keys():
-                catalog_selected_objects = np.array(par_dic['catalog_selected_objects'].split(','), dtype=np.int)
+                try:
+                    catalog_selected_objects = np.array(par_dic['catalog_selected_objects'].split(','),
+                                                        dtype=np.int)
+                except:
+                    raise RequestNotUnderstood("format not valid")
             else:
                 catalog_selected_objects = None
             if 'selected_catalog' in par_dic.keys():
                 catalog_dic = json.loads(par_dic['selected_catalog'])
-                user_catalog = build_catalog(catalog_dic, catalog_selected_objects)
+                try:
+                    user_catalog = build_catalog(catalog_dic, catalog_selected_objects)
+                except ValueError as e:
+                    e_message = str(e)
+                    raise RequestNotUnderstood(e_message)
                 self.set_par('user_catalog', user_catalog)
 
+        # TODO is this case still in use?
+        if 'user_catalog_dictionary' in par_dic.keys() and par_dic['user_catalog_dictionary'] is not None:
+            if type(par_dic['user_catalog_dictionary']) == dict:
+                catalog_to_build = par_dic['user_catalog_dictionary']
+            else:
+                catalog_to_build = json.loads(par_dic['selected_catalog'])
+            if catalog_to_build is not None:
+                try:
+                    catalog_dic = build_catalog(catalog_to_build)
+                except RuntimeError:
+                    raise RequestNotUnderstood("catalog format not valid")
+                self.set_par('user_catalog', catalog_dic)
```

Comment:
and also this?

Reply:
there are several ways of what frontend passes. It's good to verify, but it is not, in principle, up to the user. But ok.

Reply:
to me, it's not entirely clear how the `user_catalog_dictionary` should be used and when.

it's not used from the frontend though, so I suppose it's some legacy code

Reply:
possibly

Reply:
Just for the record, I made an advanced search and found out that `user_catalog_dictionary` was used in the past in some unit test, and nowhere else. So, we can get ride of it.